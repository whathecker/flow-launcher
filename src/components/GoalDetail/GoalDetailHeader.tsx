import React from "react";
import { Image, StyleSheet } from "react-native";
import { View, Text } from "../Themed";
import { CloseIcon } from "../shared";
import { navigationRef } from "../../utils";
import { Container, Typography } from "../../styles";

type GoalDetailHeaderProps = {
  title: string;
  movitation: string;
};

const GoalDetailHeader: React.FC<GoalDetailHeaderProps> = ({
  title,
  movitation,
}: GoalDetailHeaderProps) => {
  return (
    <>
      <View
        style={{
          ...Container.flexStart,
          height: "55%",
        }}
      >
        <View style={styles.closeIconWrapper}>
          <CloseIcon
            pressHandler={() => {
              navigationRef.goBack();
            }}
          />
        </View>
      </View>
      <View style={{ paddingLeft: "8%" }}>
        <View style={styles.headerWrapper}>
          <Text style={styles.goalTitle}>{title}</Text>
          <Image
            style={styles.rocketImage}
            source={require(`../../../assets/images/rocket_1f680.png`)}
          />
        </View>
        <View style={styles.motivationWrapper}>
          <Text style={styles.motivationText}>{movitation}</Text>
        </View>
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  closeIconWrapper: {
    paddingLeft: 25,
    paddingTop: 0,
  },
  headerWrapper: {
    ...Container.flexStart,
    width: "100%",
    paddingTop: 5,
    paddingBottom: 5,
  },
  rocketImage: {
    width: 32,
    height: 32,
  },
  goalTitle: {
    ...Typography.h1,
    marginRight: 15,
  },
  motivationWrapper: {
    paddingTop: 5,
    paddingBottom: 5,
  },
  motivationText: {
    ...Typography.p,
  },
});

export default GoalDetailHeader;
