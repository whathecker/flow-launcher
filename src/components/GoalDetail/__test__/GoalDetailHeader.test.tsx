import * as React from "react";
import renderer from "react-test-renderer";
import GoalDetailHeader from "../GoalDetailHeader";

describe("Test GoalDetailHeader component", () => {
  it("Should renders GoalDetailHeader correctly", () => {
    const tree = renderer
      .create(
        <GoalDetailHeader
          title="Testing the component"
          movitation="Want to get this right"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
