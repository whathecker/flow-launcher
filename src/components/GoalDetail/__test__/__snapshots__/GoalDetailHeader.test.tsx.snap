// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GoalDetailHeader component Should renders GoalDetailHeader correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "paddingBottom": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "55%",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "paddingLeft": 25,
            "paddingTop": 0,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "paddingLeft": "8%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "paddingBottom": 5,
            "paddingTop": 0,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#554F4F",
            },
            Object {
              "fontFamily": "Lexend-SemiBold",
              "fontSize": 20,
              "marginRight": 15,
            },
          ]
        }
      >
        Testing the component
      </Text>
      <Image
        source={1}
        style={
          Object {
            "height": 32,
            "width": 32,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
          },
          Object {
            "paddingBottom": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#554F4F",
            },
            Object {
              "fontFamily": "Lexend-Regular",
              "fontSize": 16,
            },
          ]
        }
      >
        Want to get this right
      </Text>
    </View>
  </View>
</View>
`;
